doc/sphinx/language/extensions/evars.rst

```diff
@@ -74,18 +74,51 @@ Expressions often contain redundant pieces of information. Subterms that can be
 automatically inferred by Coq can be replaced by the symbol ``_`` and Coq will
 guess the missing piece of information.
 
+Automatic resolution of existential variables
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Some tactics automatically resolve goals that are created when existential
+variables are created.
+
+.. _automatic-evar-resolution:
+
+.. example:: Automatic resolution of existential variables
+
+   .. coqtop:: reset in
+
+      Goal forall (a : nat), a = a -> exists n : nat, n = n.
```

Comment:
Sure, I use `Goal` because I don't spend enough time thinking of good names. Here the lemma name could be `le_not_lt` or something like that. Note that there are many places in the refman that use `Goal` and should also be changed.

Reply:
If you guys want to change all the `Goal`s to `Lemma`s, I won't stand in your way.  But I think there are lots of more important improvements to make to the doc.  Coming up with good names can be nontrivial.  We should keep the examples succinct so it's easy to pick out what's important.